waves-scripts

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+
+start_dev_env () {
+    set -e
+
+    start_nodes
+
+    local usdt_asset_id=$(mint_usdt)
+    start_bobtimus usdt_asset_id
+
+    start_frontend
+}
+
+stop_dev_env () {
+    stop_bobtimus
+    stop_nodes
+}
+
+start_nodes () {
+    set -e
+
+    # start elementsd, esplora (et al)
+    nigiri start --liquid
+}
+
+mint_usdt () {
+    set -e
+
+    local -r amount=1000000
+    address=$(nigiri rpc --liquid getnewaddress)
+
+    # need to wait a couple of seconds so that we don't fail on the second request
+    sleep 2
+
+    response=$(nigiri mint $address $amount "Tether USD" "USDt")
+    asset_id=$(echo $response | sed -n -e 's/^.*\(asset: \)\(.*\)\( issuance_txin: .*$\)/\2/p')
+
+    echo "$asset_id"
+}
+
+start_bobtimus () {
+    if [ $# -ne 1 ]; then
+        echo "Incorrect number of arguments"
+        exit 1
+    fi
+
+    if [ -f ".bobtimus" ]; then
+        echo "Are you sure Bobtimus is not already up?"
+        exit 1
+    fi
+
+    set -e
+
+    __ensure_log_dir
+    __read_env_file
+
+    local -r elementsd_rpc_user="admin1"
+    local -r elementsd_rpc_password="123"
+    RUST_LOG=info cargo run --bin fake_bobtimus -- \
+            --elementsd http://$elementsd_rpc_user:$elementsd_rpc_password@127.0.0.1:$LIQUID_NODE_PORT \
+            --usdt $1 > $log_dir/bobtimus 2>&1 &
+    bobtimus_pid=$!
+
+    # TODO: Make sure that bobtimus was actually started up successfully before recording its pid
```

Comment:
curl has a flag to make sure the request is a 2xx, you can use that to ping bobtimus on the rate endpoint and check if it is up!